kernel/kernel-config-4.9.x

```diff
@@ -161,6 +161,7 @@ CONFIG_FAIR_GROUP_SCHED=y
 # CONFIG_RT_GROUP_SCHED is not set
 # CONFIG_CGROUP_PIDS is not set
 CONFIG_CGROUP_FREEZER=y
+# CONFIG_CGROUP_HUGETLB is not set
```

Comment:
This (and the other new options added below) suggests the kernel version changed?

Reply:
@jodh-intel  -- no these are options which appear since HUGETLB is being enabled.  Similar for the UIO ones below.

Reply:
> ... which appear ...

Ugh. So, the kernel config comprises:

- explicitly set options
- explicitly unset options
- ...and a bunch of invisible options which magically appear when you explicitly set some of the previously explicitly unset options.

In summary: the kernel config shows you everything you set, and some of what you didn't. Nice.

Reply:
pretty much @jodh-intel , yes - you cannot tell in the .config itself what was set or unset by what. You can sometimes tease some of the info out from staring at the config UIs and the Kconfig files, but yes, it is a twisty maze I'm afraid. The kernel folks are aware of the limitations, but I somehow think it won't be changing any time soon.